components/admin_console/permission_schemes_settings/permission_schemes_settings.jsx

```diff
@@ -62,9 +60,9 @@ export default class PermissionSchemesSettings extends React.PureComponent {
                 promises.push(this.props.actions.loadSchemeTeams(scheme.id));
             }
             Promise.all(promises).then(() => this.setState({loading: false, phase2MigrationIsComplete}));
```

Comment:
qq: what happens if setState throws? Is it being caught by the outer catch()? According to this: https://twitter.com/dan_abramov/status/770914625206583296 ; "then(a).catch(b) makes errors from a go to b". Are we ok with it here (and if yes, should phase2MigrationIsComplete be true?).

Reply:
This replicates the same behavior than the previous one, I think we can create another ticket here to address the possible swallowing of failures in the `Promise.all().then` line, probably we should catch and decide what to do there. But I think is not in the scope of this PR because this is about the migration of `UNSAFE` methods.

@catalintomai What do you think about creating another ticket for better error handling on this components and keep this PR only for the migration from `UNSAFE` methods?

Reply:
Sure, will do.